### PR TITLE
Port the Python layer to Python 3

### PR DESCRIPTION
**Pull request by fedde-s**
Python 2 has been deprecated for a decade now, and will reach its end of life in less than 1½ years. This PR ports the Python scripts we still document and support (`metaImport`, `validateStudies` and `migrate_db`) to Python 3.4, which is included even in the older long-term-support releases of Debian and Ubuntu that haven't already reached their end of life.

Main changes proposed in this pull request:
- Port the Python layer of the data import pipeline to Python 3.4+
- Make the automated tests for the ported layer run reproducibly across minor releases of Python 3.4+
- Port database migration script to Python 3.x
- Configure Travis CI to test the data import pipeline on Python 3.4, including validateStudies

# Checks
- [ ] Runs on Heroku.
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Google Style Guide](https://github.com/google/styleguide).
- [x] If this is a feature, the PR is to rc. If this is a bug fix, the PR is to master.

### CODE REVIEW

**Comment by fedde-s** [comment]
@inodb, do you approve of this, or know of anyone who might have some input?

File: docs/Updating-your-cBioPortal-installation.md [hunk 1]
```diff
@@ -30,24 +30,20 @@ Compile your code again. After restarting the webserver the page should now stat
...
 ```
 brew install mysql-connector-c
-sudo pip2 install MySQL-python
-```
-
-For Mac OS X, you must also add the following to your .bash_profile:
-
-```
-export DYLD_LIBRARY_PATH=/usr/local/mysql/lib:$DYLD_LIBRARY_PATH
+sudo python3 -m pip install mysqlclient
```
Comment (inodb) [inline docs/Updating-your-cBioPortal-installation.md]: the `DYLD_LIBRARY_PATH=/usr/local/mysql/lib:$DYLD_LIBRARY_PATH` is no longer necessary for pyhton3 ?

Comment (fedde-s) [reply docs/Updating-your-cBioPortal-installation.md]: It wasn't clear to me where this recommendation came from in the first place, and whether it applies to modern versions of macOS and the library. I instead linked to the documentation of the library itself a few lines down, which states that some default configuration *may*(?) be broken on macOS, and recommends an entirely different fix.

**Comment by fedde-s** [comment]
@n1zea144, @angelicaochoa and @sheridancbio, can your silence so far be taken to mean that you have no objections? Or is reviewing this pull request still on your task list?

**Comment by fedde-s** [comment]
I heard back from @angelicaochoa; she said she did intend to take another look at this. Feel free to self-unassign if you feel everything you could have contributed to the review has already been said though! :)

**Comment by fedde-s** [comment]
Rebased on the last commit of `rc` that I've seen passing the tests.

**Comment by fedde-s** [comment]
Restarting the Travis build in case the mangled order of plots in the the study view screenshot test is intermittent.

**Comment by fedde-s** [comment]
If we do merge this change to `rc`, the validator bugfixes being merged to master will be written in Python 2 while the other branches have a validator written in Python 3. This would make merging `master` to `release-*` and `rc` a bit painful. Could we instead make a new major release with only this feature (and all bugfixes from `master) on top of v1.15.0?

I created the release branch for that here: https://github.com/cBioPortal/cbioportal/commits/release-1.16.0

Here's an illustration of the commit graph for that release branch, in case that explains its better than my words:
```
*   cc4504dd5 (release-1.16.0) Merge rebased pull request #4603 from thehyve/python3-importer-rebased
|\
| * bbdd496b1 (python3-importer-rebased) Add test suite for validateStudies to Travis CI
| […]
| * e386ff519 Automatically translate importer module to Python3
|/
*   a2c7072e7 Merge branch 'master' into release-1.16.0
|\
| *   0014cc2ed (master) Merge pull request #4677 from cBioPortal/jjgao-patch-1
| |
| […]
| |
* | 7ee3ca61d v1.16.0-SNAPSHOT
* |   907c01800 Automated master to rc merge
|\ \
| |/
| *   8c7725e68 Merge pull request #4598 from cBioPortal/release-1.15.0
| |\
| | * 3ef4fbf7b (upstream/release-1.15.0) v1.15.1-SNAPSHOT
| | * d96c6acfe (tag: v1.15.0) v1.15.0
```

**Comment by fedde-s** [comment]
@inodb and @n1zea144, do you think we can release this, say, today or tomorrow, and merge `release-1.16.0` as proposed above to `master`?

**Comment by fedde-s** [comment]
For clarity on GitHub, I explicitly pointed this PR to the `release-1.16.0` branch.

**Comment by fedde-s** [comment]
@inodb pointed out two more changes that should probably be included in the same release: updating the Python version listed in the Dockerfile in this repo, and giving a helpful message instead of a crash if someone does accidentally run one of the scripts with Python 2.

**Comment by fedde-s** [comment]
I'm not sure if it's feasible to print a friendly message when the scripts are being parsed as Python 2 interpreter, @inodb. Syntax errors in print functions and try-except blocks stop execution while the interpreter parses the file, before it can execute anything.

Implementing that temporary convenience might require a major architectural overhaul in which all script entrypoints are replaced by wrappers that can be parsed equally well in multiple languages, which is not something I'll be able to implement any time soon.

Can we merge this without it?

**Pull request merged**